business/istio_validations.go

```diff
@@ -408,15 +408,8 @@ func (in *IstioValidationsService) fetchDetails(rValue *kubernetes.IstioDetails,
 		errChan2 := make(chan error, 5)
 		istioDetails := kubernetes.IstioDetails{}
 
-		if IsResourceCached(namespace, kubernetes.VirtualServices) {
```

Comment:
I'd probably leave this method as it is. Because the VS/DR fetched in here are the base of the validations (at namespace level). If you modify this, then you might end up validating VS/DR that are not in the namespace but they are accessible from the namespace.

I propose you to create a new method just fetching VS/DR/ServiceEntries that are available for the ns of the validation scope. The resulting arrays of VS/DR/SE will be sent in each checker validation.

For example:

```golang
go in.fetchExportedResources(&exportedResources, namespace, errChan, &wg)

checkers.VirtualServiceChecker{Namespace: namespace, Namespaces: namespaces, DestinationRules: exportedResources.DestinationRules, VirtualServices: istioDetails.VirtualServices, ExportedVirtualServices: exportedResources.VirtualServices}

checkers.DestinationRulesChecker{Namespaces: namespaces, DestinationRules: istioDetails.DestinationRules, MTLSDetails: mtlsDetails, ServiceEntries: exportedResource.ServiceEntries}
```

Note that the checkers might need to have an extra field for the exported Resources or not. Depending on each validation. Let's take `VirtualServiceCheckers` for instance. If it contains a checker that needs to compare each virtual services with other virtual services, then we would need to compare the Virtual Service (subject of the validation) of the namespace with all the virtual services exported to that namespace. Does it make sense?

Reply:
+1